lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp

```diff
@@ -2522,7 +2516,7 @@ StringRef CppEmitter::getMemRefDimParam(Value memref, unsigned index) {
 /// associated with it
 bool CppEmitter::isMemRefDimParam(Value memref, unsigned index) {
   assert([&] {
-    MemRefType type = memref.getType().dyn_cast<MemRefType>();
+    auto type = memref.getType().dyn_cast<MemRefType>();
```

Comment:
Since we're changing this line, can we fix the postfixed use of `dyn_cast`? I believe it's being deprecated.